### PR TITLE
fix(ipc/subprocess)  

### PR DESCRIPTION
**Pull request by cirospaciari**
### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?
Existing tests

### CODE REVIEW

**Comment by robobun** [comment]
@cirospaciari, your commit e6c61ecc81ac36388d269321f72c2a77d37a7d88 is building: [`#2051`](https://buildkite.com/bun/bun/builds/2051)

File: src/bun.js/api/server.zig [hunk 1]
```diff
@@ -6011,7 +6013,9 @@ pub fn NewServer(comptime NamespaceType: type, comptime ssl_enabled_: bool, comp
                 listener.close();
             } else if (!this.flags.terminated) {
                 this.flags.terminated = true;
-                this.app.close();
+                if (this.app) |app| {
```
Comment (Jarred-Sumner) [inline src/bun.js/api/server.zig]:
```suggestion
                if (this.app) |app| {
                   this.app = null;
```

File: src/bun.js/api/server.zig [hunk 2]
```diff
@@ -6034,7 +6038,7 @@ pub fn NewServer(comptime NamespaceType: type, comptime ssl_enabled_: bool, comp
 
             if (!this.flags.terminated) {
                 this.flags.terminated = true;
-                this.app.close();
+                if (this.app) |app| app.close();
```
Comment (Jarred-Sumner) [inline src/bun.js/api/server.zig]:
```suggestion
                if (this.app) |app| {
                  this.app = null;
                  app.close();
                }
```

Comment (cirospaciari) [reply src/bun.js/api/server.zig]: we still need to call destroy on it.

File: src/bun.js/ipc.zig [hunk 3]
```diff
@@ -537,19 +537,19 @@ const NamedPipeIPCData = struct {
...
+        JSC.VirtualMachine.get().enqueueTask(JSC.ManagedTask.New(NamedPipeIPCData, closeTask).init(this));
     }
 
     pub fn closeTask(this: *NamedPipeIPCData) void {
         log("NamedPipeIPCData#closeTask", .{});
         if (this.disconnected) {
-            this.writer.close();
-            this.onClientClose();
+            if (this.server) |server| {
+                server.close(onServerClose);
```
Comment (Jarred-Sumner) [inline src/bun.js/ipc.zig]:
```suggestion
                this.server = null;
                server.close(onServerClose);
```

File: src/bun.js/api/bun/subprocess.zig [hunk 4]
```diff
@@ -2319,8 +2326,12 @@ pub const Subprocess = struct {
     pub fn handleIPCClose(this: *Subprocess) void {
         IPClog("Subprocess#handleIPCClose", .{});
         this.updateHasPendingActivity();
-        const ok = this.ipc_data != null;
-        if (ok) this.ipc().internal_msg_queue.deinit();
+        defer this.deref();
```
Comment (Jarred-Sumner) [inline src/bun.js/api/bun/subprocess.zig:2336]: only if `ipc_data` is not null right?

Comment (cirospaciari) [reply src/bun.js/api/bun/subprocess.zig:2336]: we always need to deref here because always we ref earlier

File: src/bun.js/ipc.zig (hunk 3 above)
Comment (cirospaciari) [reply src/bun.js/ipc.zig]: we free it inside `onServerClose`

**Pull request merged**